src/black/linegen.py

```diff
@@ -142,6 +142,14 @@ def visit_stmt(
 
             yield from self.visit(child)
 
+    def visit_match_case(self, node: Node) -> Iterator[Line]:
+        """Visit either a match or case statement."""
+        normalize_invisible_parens(node, parens_after=set())
+
+        yield from self.line()
```

Comment:
I am not sure if this is the correct solution; it originates from the assumptions that for each `match_statement`/`case_statement` node, there will be only one valid `match`/`case` keyword for the context. So we simply yield a line; and for all following encounters of child nodes, we process them as usual.